File: src/uiMain/funcionalidades/Asignar.java
/*
 * Clase que permite asignar un tiquete a un Comprador, un vehiculo a un especialista 
 * Estructuras: ArrayList
 * 
 * @author: Mateo Hechavarria, Juan Pablo Pineda, Miguel Angel Fonseca, Haison Urrutia
 */



package uiMain.funcionalidades;

import gestorAplicacion.*;

import java.io.Serializable;
import java.time.LocalDate;
import java.util.ArrayList;

public class Asignar implements Serializable {
    public static Tiquete asignarTiquete(Comprador comprador, Tiquete tiquete){
        tiquete.setComprador(comprador);
        tiquete.setFechaCompra(LocalDate.now());
        tiquete.setEstado(true);
        comprador.anadirTiqueteHistoria(tiquete);
        tiquete.getViaje().getDestino().anadirVisitantes(1);
        return tiquete;
    }
    public static Viaje asignarVehiculo(Conductor conductor, Viaje viaje){
        viaje.setConductor(conductor);
        conductor.anadirViajeHistoria(viaje);
        return viaje;
    }
    public static String asignarVehiculo(Especialista mecanico, Vehiculo vehiculo){
        mecanico.anadirVehiculoHistoria(vehiculo);
        return mecanico.revisionVehiculo(vehiculo);
    }
}

File: Python/src/uiMain/Funcionalidades/Asignar.py
from datetime import datetime
from gestorAplicacion.personas.Comprador import Comprador
from gestorAplicacion.viajes.Tiquete import Tiquete
from gestorAplicacion.viajes.Vehiculo import Vehiculo
from gestorAplicacion.viajes.Viaje import Viaje
from gestorAplicacion.personas.Conductor import Conductor
from gestorAplicacion.personas.Especialista import Especialista

class Asignar():
    '''
    Asignar : Contiene la informacion de: 

    Esta clase tiene como objetivo implementar los metodos se asignacion a Empleado y Comprador para ser llamado
    desde las vemntanas o desde las  funcionalidades
    '''
    
    @staticmethod
    def asignarTiquete(comprador : Comprador, tiquete : Tiquete):
        tiquete.setComprador(comprador)
        tiquete.setFechaCompra(datetime.now())
        tiquete.setEstado(True)
        comprador.anadirTiqueteHistoria(tiquete)
        tiquete.getViaje().getDestino().setNumVisitantes(1)
    
    @staticmethod
    def asignarVehiculoConductor(conductor : Conductor, viaje : Viaje):
        viaje.setConductor(conductor)
        conductor.anadirViajeHistoria(viaje)
    
    @staticmethod
    def asignarVehiculoEspecialista(especialista : Especialista, vehiculo : Vehiculo):
        especialista.anadirVehiculoHistoria(vehiculo)
